eZ/Publish/Core/Repository/Tests/Service/Mock/ContentTest.php

```diff
@@ -3170,6 +3173,109 @@ public function testUpdateContentThrowsUnauthorizedException()
+                'languageCode' => 'fre-FR',
+                'fieldDefIdentifier' => 'name',
+                'fieldTypeIdentifier' => 'ezstring',
+            ]
+        );
+        $updatedField2 = new Field(
+            [
+                'value' => new TextLineValue('two'),
+                'languageCode' => 'fre-FR',
+                'fieldDefIdentifier' => 'name',
+                'fieldTypeIdentifier' => 'ezstring',
+            ]
+        );
+
+        $contentUpdateStruct = new ContentUpdateStruct([
+            'initialLanguageCode' => 'fre-FR',
+            'fields' => [
+                $updatedField,
+                $updatedField2,
+            ],
+        ]);
+
+        $versionInfo = new VersionInfo(
+            [
+                'contentInfo' => new ContentInfo(['id' => $contentId, 'mainLanguageCode' => 'eng-GB']),
+                'versionNo' => $versionNo,
+                'status' => APIVersionInfo::STATUS_DRAFT,
+            ]
+        );
+
+        $content = new Content(
+            [
+                'versionInfo' => $versionInfo,
+                'internalFields' => [
+                    new Field(
+                        [
+                            'value' => new TextLineValue('one'),
+                            'languageCode' => 'eng-GB',
+                            'fieldDefIdentifier' => 'name',
+                            'fieldTypeIdentifier' => 'ezstring',
+                        ]
+                    ),
+                    $updatedField2,
+                ],
+                'contentType' => new ContentType([
+                    'fieldDefinitions' => new FieldDefinitionCollection([
+                        new FieldDefinition([
+                            'identifier' => 'name',
+                            'fieldTypeIdentifier' => 'ezstring',
+                        ]),
+                    ]),
+                ]),
+            ]
+        );
+
+        $mockedService->expects(self::once())
+            ->method('loadContent')
+            ->with($contentId, null, $versionNo)
+            ->willReturn($content);
+
+        $this->getFieldTypeRegistryMock()->expects(self::any())
+            ->method('getFieldType')
+            ->willReturn(new Type());
+
+        $permissionResolverMock->expects(self::once())
+            ->method('canUser')
+            ->with(
+                'content',
+                'edit',
+                $content,
+                [
+                    (new VersionBuilder())
+                        ->updateFields([
+                            $updatedField,
+                        ])
+                        ->updateFieldsTo(
+                            $contentUpdateStruct->initialLanguageCode,
+                            $contentUpdateStruct->fields
+                        )
+                        ->build(),
+                ]
+            )
+            ->willReturn(false);
+
+        $this->expectException(UnauthorizedException::class);
```

Comment:
wasn't the issue that you're actually getting an exception and you shouldn't?

Reply:
Please, see above.